src/Features/CSharp/Portable/SignatureHelp/AbstractCSharpSignatureHelpProvider.LightweightOverloadResolution.cs

```diff
@@ -5,297 +5,252 @@
 using System;
 using System.Collections.Immutable;
 using System.Diagnostics;
-using System.Diagnostics.CodeAnalysis;
 using Microsoft.CodeAnalysis.CSharp.Extensions;
 using Microsoft.CodeAnalysis.CSharp.Syntax;
-using Microsoft.CodeAnalysis.LanguageService;
-using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Collections;
 using Microsoft.CodeAnalysis.Shared.Extensions;
 using Roslyn.Utilities;
 
-namespace Microsoft.CodeAnalysis.CSharp.SignatureHelp
+namespace Microsoft.CodeAnalysis.CSharp.SignatureHelp;
+
+internal abstract partial class AbstractCSharpSignatureHelpProvider
 {
-    internal abstract partial class AbstractCSharpSignatureHelpProvider
+    protected readonly struct LightweightOverloadResolution(
+        SemanticModel semanticModel,
+        int position,
+        SeparatedSyntaxList<ArgumentSyntax> arguments)
```

Comment:
made this a helper struct.  moved up common, readonly, state to the primary constructor.